examples/rag/rag_agents.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+"""
+This example shows how to build a agent with RAG (backup by LlamaIndex)
+"""
+
+from typing import Optional
+from llama_index.embeddings.dashscope import DashScopeEmbedding
+from llama_index.core import Settings, VectorStoreIndex, SimpleDirectoryReader
+
+from agentscope.prompt import PromptType
+from agentscope.agents.agent import AgentBase
+from agentscope.prompt import PromptEngine
+from agentscope.message import Msg
+from agentscope.rag.llama_index_rag import LlamaIndexRAG
+
+
+Settings.embed_model = DashScopeEmbedding()
+
+
+class LlamaIndexAgent(AgentBase):
```

Comment:
Shall we name it `RAGAgent` and add a parameter `backend_type` to indicate whether to use `LlamaIndexRAG` or `LangChainRAG`?

Reply:
Personally, I would incline to seperate llama-index and langchain implementation. Because their dependencies are very different, and we may expect our user only use one of those and don't need to be bother about the dependencies of the other.